commit a5398181a98479dc99e80e451f434928402c5fdf
Author: Denis Barbier <bouzim@gmail.com>
Date:   Thu May 5 00:46:58 2011 +0200

    Fix build failures with g++ 4.6
    
    Error message is:
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx: In static member function 'static Standard_Boolean Visual3d_TransientManager::BeginDraw(const Handle_Visual3d_View&, Standard_Boolean, Standard_Boolean)':
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx:123:70: error: taking address of temporary [-fpermissive]
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx:127:68: error: taking address of temporary [-fpermissive]
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx: In static member function 'static void Visual3d_TransientManager::ClearDraw(const Handle_Visual3d_View&, Standard_Boolean)':
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx:192:77: error: taking address of temporary [-fpermissive]
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx:194:75: error: taking address of temporary [-fpermissive]
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx: In static member function 'static Standard_Boolean Visual3d_TransientManager::BeginAddDraw(const Handle_Visual3d_View&)':
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx:232:79: error: taking address of temporary [-fpermissive]
     ../../../../ros/src/Visual3d/Visual3d_TransientManager.cxx:234:77: error: taking address of temporary [-fpermissive]
    
    Submitted upstream: http://www.opencascade.org/org/forum/thread_20572/

Index: oce/src/Visual3d/Visual3d_TransientManager.cxx
===================================================================
--- oce.orig/src/Visual3d/Visual3d_TransientManager.cxx
+++ oce/src/Visual3d/Visual3d_TransientManager.cxx
@@ -120,11 +120,11 @@ Aspect_CLayer2d UnderCLayer;
 
 	if (! UnderLayer.IsNull ()){
 		UnderCLayer = UnderLayer->CLayer ();
-		theCView.ptrUnderLayer = (CALL_DEF_LAYER *) &(UnderLayer->CLayer ());
+		theCView.ptrUnderLayer = (CALL_DEF_LAYER *) &UnderCLayer;
 	}
 	if (! OverLayer.IsNull ()){
 		OverCLayer = OverLayer->CLayer ();
-		theCView.ptrOverLayer = (CALL_DEF_LAYER *) &(OverLayer->CLayer ());
+		theCView.ptrOverLayer = (CALL_DEF_LAYER *) &OverCLayer;
 	}
 
 	// Begin rendering
@@ -188,10 +188,16 @@ void Visual3d_TransientManager::ClearDra
 
 	// Begin rendering
 	theCView	= *(CALL_DEF_VIEW *)AView->CView ();
-	if (!AView->UnderLayer().IsNull()) 
-		theCView.ptrUnderLayer = (CALL_DEF_LAYER *)&(AView->UnderLayer()->CLayer());
-	if (!AView->OverLayer().IsNull()) 
-		theCView.ptrOverLayer = (CALL_DEF_LAYER *)&(AView->OverLayer()->CLayer());
+Aspect_CLayer2d OverCLayer;
+Aspect_CLayer2d UnderCLayer;
+	if (!AView->UnderLayer().IsNull()) {
+		UnderCLayer = AView->UnderLayer()->CLayer ();
+		theCView.ptrUnderLayer = (CALL_DEF_LAYER *)&UnderCLayer;
+	}
+	if (!AView->OverLayer().IsNull()) {
+		OverCLayer = AView->OverLayer()->CLayer();
+		theCView.ptrOverLayer = (CALL_DEF_LAYER *)&OverCLayer;
+	}
 
 	Handle(Aspect_GraphicDriver) agd = AView->GraphicDriver ();
 
@@ -228,10 +234,16 @@ Standard_Boolean Visual3d_TransientManag
 
 	// Begin rendering
 	theCView	= *(CALL_DEF_VIEW *)AView->CView ();
-	if (!AView->UnderLayer().IsNull()) 
-		theCView.ptrUnderLayer = (CALL_DEF_LAYER *) &(AView->UnderLayer()->CLayer ());
-	if (!AView->OverLayer().IsNull()) 
-		theCView.ptrOverLayer = (CALL_DEF_LAYER *) &(AView->OverLayer()->CLayer ());
+Aspect_CLayer2d OverCLayer;
+Aspect_CLayer2d UnderCLayer;
+	if (!AView->UnderLayer().IsNull()) {
+		UnderCLayer = AView->UnderLayer()->CLayer ();
+		theCView.ptrUnderLayer = (CALL_DEF_LAYER *)&UnderCLayer;
+	}
+	if (!AView->OverLayer().IsNull()) {
+		OverCLayer = AView->OverLayer()->CLayer();
+		theCView.ptrOverLayer = (CALL_DEF_LAYER *)&OverCLayer;
+	}
 
 	Handle(Aspect_GraphicDriver) agd = AView->GraphicDriver ();
 
